src/Features/LanguageServer/Protocol/Handler/Completion/CompletionResolveHandler.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.LanguageServer.CustomProtocol;
+using Microsoft.VisualStudio.Text.Adornments;
+using Newtonsoft.Json.Linq;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    /// <summary>
+    /// Handle a completion resolve request to add description.
+    /// </summary>
+    [Shared]
+    [ExportLspMethod(LSP.Methods.TextDocumentCompletionResolveName)]
+    internal class CompletionResolveHandler : IRequestHandler<LSP.CompletionItem, LSP.CompletionItem>
+    {
+        public async Task<LSP.CompletionItem> HandleRequestAsync(Solution solution, LSP.CompletionItem completionItem,
+            LSP.ClientCapabilities clientCapabilities, CancellationToken cancellationToken)
+        {
+            CompletionResolveData data;
+            if (completionItem.Data is CompletionResolveData)
```

Comment:
not sure how this situation can happen. workitem for us to take a look later?

Reply:
or to add test to cover both case.

Reply:
https://github.com/dotnet/roslyn/projects/45#card-20866454